tc/sgx/trusted_worker_manager/enclave/iohandler_enclave.cpp

```diff
@@ -0,0 +1,58 @@
+/* Copyright 2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "enclave_t.h"
+#include "tcf_error.h"
+#include "error.h"
+#include "iohandler_enclave.h"
+#include "enclave_utils.h"
+
+uint32_t TcfExecuteIoCommand(uint32_t handlerId,
+                             const uint8_t* command,
+                             size_t commandSize,
+                             uint8_t* result,
+                             size_t resultSize,
+                             const uint8_t* inBuf,
+                             size_t inBufSize,
+                             uint8_t* outBuf,
+                             size_t outBufSize)
+{
+    uint32_t status = 0;
+    tcf::error::ThrowIfNull(command, "IO command is null");
+    try
+    {
+        status = ocall_Process(&status, handlerId, (const char*) command,
+            commandSize, result, resultSize, inBuf, inBufSize, outBuf, outBufSize);
+    } catch (tcf::error::Error& e) {
+        ocall_SetErrorMessage(e.what());
+        status = e.error_code();
+    } catch (...)
+    {
+        status = TCF_ERR_UNKNOWN;
+    }
+
+    return 0;
+}
+
+uint32_t TcfGetIoHandlerId(const char* handlerName)
+{
+    if (handlerName == "tcf-base-file-io") {
```

Comment:
This should be later changed to lookup from io-handler registry

Reply:
Agree.